apps/api/src/campaign/campaign.service.ts

```diff
@@ -713,8 +720,69 @@ export class CampaignService {
         },
       })
     }
+    // Create a notification list for this campaign when updated to active
+    if (
+      campaign?.state !== CampaignState.active &&
+      updated?.state === CampaignState.active &&
+      // No list exists yet
+      !campaign?.notificationLists?.length
+    )
+      try {
+        await this.createCampaignNotificationList(updated)
+      } catch (e) {
+        Logger.error('Failed to create notification list', e)
+      }
 
-    return result
+    // Update notification list name with the new campaign name
+    if (
+      updated?.state === CampaignState.active &&
+      campaign?.notificationLists?.length &&
+      updated.title !== campaign.notificationLists[0]?.name
+    )
+      try {
+        // Get the list
+        const list = campaign?.notificationLists[0]
+        if (list) await this.updateCampaignNotificationList(updated, list)
+      } catch (e) {
+        Logger.error('Failed to update notification list', e)
+      }
+
+    return updated
+  }
+
+  async createCampaignNotificationList(updated: { title: string; id: string }) {
+    // Generate list in the marketing platform
+    const listId = await this.generalNotificationsService.createNewContactList({
+      name: updated.title || updated.id,
+    })
+    // Save the list_id in the DB
+    await this.prisma.notificationList.create({
+      data: {
+        id: listId,
+        name: updated.title,
+        campaignId: updated.id,
+      },
+    })
+
+    return listId
+  }
+
+  async updateCampaignNotificationList(
+    updated: { title: string; id: string },
+    list: NotificationList,
+  ) {
+    // Update list name in the Marketing Platform
+    await this.generalNotificationsService.updateContactList({
+      data: { name: updated.title },
+      id: list.id,
+    })
+    // update in DB
+    await this.prisma.notificationList.update({
+      where: { id: list.id },
+      data: {
+        name: updated.title,
+      },
+    })
   }
```

Comment:
These two methods look like specific to the NotificationService - my suggestion is to move them there